lib/template/macros.c

```diff
@@ -339,14 +339,15 @@ log_macro_expand(GString *result, gint id, gboolean escape, const LogTemplateOpt
     case M_SOURCE_IP:
       {
         gchar *ip;
+        int checker;
 
-        if (msg->saddr && (g_sockaddr_inet_check(msg->saddr) ||
+        checker = msg->saddr;
 #if SYSLOG_NG_ENABLE_IPV6
-            g_sockaddr_inet6_check(msg->saddr))
+        checker = checker && (g_sockaddr_inet_check(msg->saddr) || g_sockaddr_inet6_check(msg->saddr));
 #else
-            0)
+        checker = checker && g_sockaddr_inet_check(msg->saddr);
 #endif
-           )
+        if(checker)
           {
```

Comment:
I agree with the change in general however, we should use better naming of this variable.

Also checker = msg->saddr completely disregards typing. So apart from the name it should be a Boolean.

And if we are at it, I would try to extract the ipv4 or ipv6 check to a separate function